src/components/page-details.jsx

```diff
@@ -43,12 +43,14 @@ export default class PageDetails extends React.Component {
 
   componentWillUnmount () {
     window.removeEventListener('keydown', this);
+    window.document.querySelector('title').innerText = 'web-monitoring-ui';
   }
 
   /**
    * @param {PageDetailsProps} previousProps
    */
   componentDidUpdate (previousProps) {
+    window.document.querySelector('title').innerText = `Scanner | ${this.state.page.url}`;
```

Comment:
I feel like we should have a test for this but uncertain of where it would go. I guess we could create a `page-details` test and add a `title` element when rendering to simulate the functionality, but not sure if there is a more accurate way that doesn't require a lot of overhead. @Mr0grog thoughts?

Reply:
I would love to learn how to write a test for this but I don't have any knowledge of how to make tests yet.  Let me know if/how I can help with making a test for this though!

Reply:
Oh, here are some basics, then! We put our tests in directories named `__test__` in the folder nearest the code they’re testing, and files should be named `<name of module>.test.jsx`. See here for component tests: https://github.com/edgi-govdata-archiving/web-monitoring-ui/tree/master/src/components/__tests__

You might start by copying and modifying [the tests for `page-list`](https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/master/src/components/__tests__/page-list.test.jsx). We use [Jest](https://jestjs.io/) and [Enzyme](https://airbnb.io/enzyme/) for our tests, so read up on those a bit. (Most of what you’ll see is how you do tests with Jest; Enzyme only manages rendering React components in a manageable and easy(ish) to test way.)

Tests look roughly like:

```js
describe('some group name, usually the component you are testing', function () {
  // Each test is in a block like `it('name of test', testLogic)`
  it('should do <whatever>', function () {
    // Set up the test by creating some data, rendering a component, whatever
    rendered = shallow(<PageDetails page={someData}>);

    // Assert that something is the way it should be after doing the above:
    expect(document).toHaveProperty('title', '<title you expect to see>');
  }
}
```

You can find detailed docs on Jest’s assertions at: https://jestjs.io/docs/en/expect

Mainly, your test will need to do 5 things:

1. Load some fake data for a page. As an example, the `page-list` tests do that here: https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/4f9a73301d3d7dad08033825bca10d55c97b09cf/src/components/__tests__/page-list.test.jsx#L9-L12

    And use it here: https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/4f9a73301d3d7dad08033825bca10d55c97b09cf/src/components/__tests__/page-list.test.jsx#L25-L29

    You could also just write the sample data directly in the test instead of loading it from a mock data file, similar to how the [`source-info` test](https://github.com/edgi-govdata-archiving/web-monitoring-ui/blob/master/src/components/__tests__/source-info.test.jsx) does it.

2. Render the component.

3. Assert that the document’s title is what you expect it to be. (i.e. `Scanner | <url from your test data>`).

4. Unmount the component.

5. Assert that the document’s title no longer shows page-specific info.

Reply:
Unrelated side note: our pages always have a `<title>` tag, so you’re code is fine, but it’s possible to create an HTML page without one:

```html
<!doctype html>
<html>
  <body>Hi.</body>
</html>
```

Firefox, for example, does not *create* a title tag for the above page, so calling `window.document.querySelector('title').innerText` would break on this page.

*In general,* it’s better practice to use `document.title`, which will always work. :)

Reply:
Feel free to reach out on slack if you run into any issues!

Reply:
Awesome, okay! I will take a crack at this!

Reply:
Okay, apologies for the wait.  I created a test file for this, and it passes the tests written out.  However, it also is throwing some UnhandledPromiseRejectionWarning's when the tests are run.  I tried hunting down what might be causing the warnings and I think it is because the ShallowWrapper is not able to carry out the .setState() method being called from componentDidMount()>this._loadPage().  Any ideas on how to handle those warnings?

Reply:
[What I see after running the test](https://i.imgur.com/qQDWNVr.png)